Template Editor: Use the label 'Clear customizations' when changes are revertable

Mamaduka opened the pull request:
## What?
Resolves #40885.

PR updates the delete button label to "Clear customizations" when changes to a template are revertable (template has theme file). It also makes the button non-destructive for this case.

## Testing Instructions
Using TT2

1. Open a Post or Page.
2. Modify the default template.
3. Confirm that dropdown displays the new label.
4. Create a new template.
5. Confirm that it displays the "Delete template" label.

## Screenshots or screencast <!-- if applicable -->
![CleanShot 2022-05-09 at 16 54 04](https://user-images.githubusercontent.com/240569/167415287-ca502d87-585f-4793-887e-72d74eb7517b.png)
![CleanShot 2022-05-09 at 16 54 22](https://user-images.githubusercontent.com/240569/167415293-cf19ebc7-6805-4b7c-957a-81d4000aebf5.png)

Mamaduka committed:
Template Editor: Use the label 'Clear customizations' when changes are revertable

Mamaduka commented on packages/edit-post/src/components/header/template-title/delete-template.js:
```diff
@@ -77,14 +79,15 @@ export default function DeleteTemplate() {
-					isDestructive
+					isDestructive={ ! isRevertable }
 					variant="secondary"
-					aria-label={ __( 'Delete template' ) }
```
The component already has a text label, no need to use the aria.

Mamaduka commented on packages/edit-post/src/components/header/template-title/template-description.js:
```diff
@@ -32,7 +32,7 @@ export default function TemplateDescription() {
 				size="body"
 				as="p"
-				style={ { marginTop: '12px' } }
+				style={ { marginTop: '12px', marginBottom: '12px' } }
```
This fixed "padding" mentioned in the issue :)

jameskoster commented:
Thanks for looking in to this :) Could the button be the same as the one in the Site Editor, IE with the description? I think the description is helpful in this context, otherwise you might think that the "Customisations" refer to unsaved changes to the page itself.

Mamaduka commented:
@jameskoster, sure. I will try to do a follow-up (in this PR) later today or tomorrow.

Mamaduka committed:
Add info text and adjust styles

Mamaduka commented:
Hi, @jameskoster

I added the info text and adjusted styles a bit to match the dropdown from the Site Editor.

## Screenshot

![CleanShot 2022-05-11 at 11 08 37](https://user-images.githubusercontent.com/240569/167790198-050a36bf-2bdb-48b4-899e-3ef76cdf176a.png)

Mamaduka commented on packages/edit-post/src/components/header/template-title/style.scss:
```diff
@@ -48,8 +48,7 @@
 	.components-popover__content {
-		min-width: 280px;
-		padding: $grid-unit-10;
+		min-width: 240px;
```
This is the `min-width` used in the Site Editor.

jameskoster replied:
240px creates an orphan in the helper text beneath the Title input:

<img width="281" alt="Screenshot 2022-05-11 at 14 19 44" src="https://user-images.githubusercontent.com/846565/167859738-ca11396e-908a-4a7f-ae75-8d4b97739dbd.png">

I know it's a minor issue and only affects `en` locales, but it might be nice to avoid 🤔

Mamaduka replied:
Sure, I'm going revert this back to `280px` 👍

Mamaduka replied:
Done ✅

jameskoster committed:
Align styling with site editor

jameskoster commented:
Optimistic that it would be a small task I attempted to better align the styles with the site editor. I got there, but it took way more code than I'd have expected. It'd be really nice if both editors just used the same component :D 

| Site Editor | Template Editor | 
| --- | --- |
| <img width="302" alt="Screenshot 2022-05-11 at 12 01 19" src="https://user-images.githubusercontent.com/846565/167834683-06c57763-eb4a-452d-b8ec-ea84b7d57eea.png"> <img width="307" alt="Screenshot 2022-05-11 at 12 02 22" src="https://user-images.githubusercontent.com/846565/167834848-e3d2e8e7-a4c9-4f83-beb7-dec37b0d94eb.png"> | <img width="324" alt="Screenshot 2022-05-11 at 11 59 39" src="https://user-images.githubusercontent.com/846565/167834648-99c3dbad-91fe-4f36-b45a-c21ea6c16552.png"> <img width="311" alt="Screenshot 2022-05-11 at 11 59 46" src="https://user-images.githubusercontent.com/846565/167834653-dbbdaef8-813e-4933-b56d-adb0d8fd7233.png"> |

Mamaduka commented:
Thanks, @jameskoster.

> It'd be really nice if both editors just used the same component

I was thinking the same but considering some of the differences; I think it's okay to keep them separate for now.

Mamaduka committed:
Remove fragment

Mamaduka committed:
Restore min-width

jameskoster commented:
> I was thinking the same but considering some of the differences; I think it's okay to keep them separate for now.

Appreciate it's probably something to do separately, but they probably shouldn't be different at all, right?

Mamaduka commented:
> but they probably shouldn't be different at all, right?

Delete action is a bit different, and we probably don't want to display the permanent delete action in Site Editor. This can be abstracted, but we also want to avoid early abstractions 😅
